hyperopt/tests/test_spark.py

```diff
@@ -432,3 +435,22 @@ def fn(x):
         self.assertGreaterEqual(
             spark_trials.count_cancelled_trials(), 1,
             "Expected at least 1 cancelled trial but found none.")
+
+    def test_invalid_timeout(self):
+        with self.assertRaisesRegexp(
+                Exception,
+                "timeout argument should be None or a positive value. Given value: -1"):
+            SparkTrials(parallelism=4, timeout=-1)
+        with self.assertRaisesRegexp(
+                Exception,
+                "timeout argument should be None or a positive value. Given value: True"):
+            SparkTrials(parallelism=4, timeout=True)
+
+    def test_exception_when_spark_not_available(self):
+        import hyperopt
+        hyperopt.spark._have_spark = False
+        try:
+            with self.assertRaisesRegexp(Exception, "cannot import pyspark"):
+                SparkTrials(parallelism=4)
+        finally:
+            hyperopt.spark._have_spark = True
```

Comment:
Why setting `True` in `finally`? Does this mean whenever you run this test, you will have `hyperopt.spark._have_spark == True` after running it. Is this what we want?

Reply:
Oh you're right!  I should set it to be the original value.